src/core/lib/gprpp/single_set_ptr.h

```diff
@@ -0,0 +1,81 @@
+// Copyright 2022 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <grpc/support/port_platform.h>
+
+#include <atomic>
+
+#include <grpc/support/log.h>
+
+namespace grpc_core {
+
+template <class T>
+class SingleSetPtr {
+ public:
+  SingleSetPtr() = default;
+  ~SingleSetPtr() {
+    T* p = p_.load(std::memory_order_relaxed);
+    if (p != sentinel()) delete p;
+  }
+
+  SingleSetPtr(const SingleSetPtr&) = delete;
+  SingleSetPtr& operator=(const SingleSetPtr&) = delete;
+  SingleSetPtr(SingleSetPtr&& other) noexcept
+      : p_(other.p_.exchange(sentinel())) {}
+  SingleSetPtr& operator=(SingleSetPtr&& other) noexcept {
+    T* p = p_.exchange(other.p_.exchange(sentinel(), std::memory_order_acq_rel),
```

Comment:
`T* p` doesn't seem to be used.

Isn't this a bug? Should we call `Set(other.p_.exchange(...))` instead?

Reply:
yeah that was a bug - fixed